server/src/main/java/keywhiz/service/daos/SecretSeriesDAO.java

```diff
@@ -538,6 +493,21 @@ private static void softDeleteSecretSeries(DSLContext dslContext, SecretsRecord
         .where(SECRETS.ID.eq(record.getId()))
         .execute();
 
+    dslContext
+        .insertInto(DELETED_ACCESSGRANTS)
+        .columns(Arrays.stream(DELETED_ACCESSGRANTS.fields())
+            .filter(field -> field != DELETED_SECRETS.ID)
```

Comment:
since I'm filtering out ID here, I would expect ID to be auto-assigned. but I'm getting this error:

```
org.jooq.exception.DataAccessException: SQL [insert into `accessgrants` (`id`, `groupid`, `secretid`, `row_hmac`, `updatedat`, `createdat`) values (?, ?, ?, ?, ?, ?)]; Duplicate entry '1' for key 'PRIMARY'
```

so jooq is generating SQL that explicitly sets the ID to the value `1`